processes/code-styles/swift/swift-style-guide-en.md

```diff
@@ -1,59 +1,269 @@
+    }
+
+    /* ... */
+}
+```
+
+* **2.16** Colons **MUST** have no space on the left and one space on the right. Exceptions are the ternary operator `? :` and empty dictionary `[:]`.
+
+**Preferred:**
+
+```swift
+// specifying type
+let pirateViewController: PirateViewController
+
+// dictionary syntax (note that we left-align as opposed to aligning colons)
+let ninjaDictionary: [String: AnyObject] = [
+    "fightLikeDairyFarmer": false,
+    "disgusting": true
+]
+
+// declaring a function
+func myFunction<T, U: SomeProtocol>(firstArgument: U, secondArgument: T) where T.RelatedType == U {
+    /* ... */
+}
+
+// calling a function
+someFunction(someArgument: "Kitten")
+
+// superclasses
+class PirateViewController: UIViewController {
+    /* ... */
+}
+
+// protocols
+extension PirateViewController: UITableViewDataSource {
+    /* ... */
+}
+```
+
+```swift
+class TestDatabase: Database {
+var data: [String: CGFloat] = ["A": 1.2, "B": 3.2]
+}
+```
+
+**Not Preferred:**
+
+```swift
+class TestDatabase : Database {
+    var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
+
+* **2.17** In general, there should be a space following a comma.
+
+```swift
+let myArray = [1, 2, 3, 4, 5]
+```
+
+* **2.18** There should be a space before and after a binary operator such as `+`, `==`, or `->`. There should also not be a space after a `(` and before a `)`.
+
+```swift
+let myValue = 20 + (30 / 2) * 3
+if 1 + 1 == 3 {
+    fatalError("The universe is broken.")
+}
+func pancake(with syrup: Syrup) -> Pancake {
+    /* ... */
+}
+```
+
+* **2.19** Follow Xcode's recommended indentation style (i.e. your code should not change if CTRL-I is pressed). When declaring a function that spans multiple lines, prefer using that syntax to which Xcode, as of version 7.3, defaults.
+
+```swift
+// Xcode indentation for a function declaration that spans multiple lines
+func myFunctionWithManyParameters(parameterOne: String,
+                                  parameterTwo: String,
+                                  parameterThree: String) {
+    // Xcode indents to here for this kind of statement
+    print("\(parameterOne) \(parameterTwo) \(parameterThree)")
+}
+
+// Xcode indentation for a multi-line `if` statement
+if myFirstValue > (mySecondValue + myThirdValue)
+    && myFourthValue == .someEnumValue {
+
+    // Xcode indents to here for this kind of statement
+    print("Hello, World!")
+}
+```
+
+* **2.20** When calling a function that has many parameters, put each argument on a separate line with a single extra indentation.
+
+```swift
+someFunctionWithManyArguments(
+    firstArgument: "Hello, I am a string",
+    secondArgument: resultFromSomeFunction(),
+    thirdArgument: someOtherLocalProperty)
+```
+
+* **2.21** When dealing with an implicit array or dictionary large enough to warrant splitting it into multiple lines, treat the `[` and `]` as if they were braces in a method, `if` statement, etc. Closures in a method should be treated similarly.
```

Comment:
Implicit array or dictionary is a nested call. We must avoid it. So I think we must not use implicit arrays or other collections as arguments. Especially if a expression is multiline

Reply:
Good point. Changed to variable in example.